lightning/src/ln/channelmanager.rs

```diff
@@ -1159,6 +1159,11 @@ pub struct CounterpartyForwardingInfo {
 	/// such that the outgoing HTLC is forwardable to this counterparty. See `msgs::ChannelUpdate`'s
 	/// `cltv_expiry_delta` for more details.
 	pub cltv_expiry_delta: u16,
+	/// The smallest value HTLC (in msat) the remote peer will accept, for this channel. This field
+	/// is only None for `CounterpartyForwardingInfo` objects serialized prior to LDK 0.0.107
+	pub outbound_htlc_minimum_msat: Option<u64>,
```

Comment:
Now that we have this here we should probably delete the copy at `Channel::counterparty_htlc_minimum_msat` - its entirely redundant and somewhat confusing to have two copies of the same field in the same object.

Reply:
Sorry for the delay in continuing this PR. @TheBlueMatt I'd need some guidance on how to continue given your comment.

In the current version of LDK, the `Channel::counterparty_htlc_minimum_msat` value is a non optional field that is assigned through the `OpenChannel` msg for inbound channels in the `new_from_req` function, and through the `AcceptChannel` msg in the `accept_channel` function in for outbound channels. The new `Channel::counterparty_forwarding_info::outbound_htlc_minimum_msat` would be an Optional field that's assigned at a later stage in the message flow.

The `Channel::counterparty_forwarding_info` field is first assigned through the first `ChannelUpdate` msg in the `channel_update` function, which with this PR will assign the new `outbound_htlc_minimum_msat` field.
The `channel_update` function never updates the value for the `Channel::counterparty_htlc_minimum_msat` field in the current version of LDK, which already proves your point about why it's a bad idea two have two copies of the same field, even though they should be the same value in practice.

I can't really assign the `Channel::counterparty_forwarding_info::outbound_htlc_minimum_msat` field through the info in the `OpenChannel` msg or the `AcceptChannel` msg, as the rest of the information to construct the `Channel::counterparty_forwarding_info` field doesn't exist yet, unless I assign dummy values to the other fields of the `CounterpartyForwardingInfo`.

**Given the above, how do we want to proceed?**
1. Simply ignoring the values for `htlc_minimum_msat` in the `OpenChannel` and `AcceptChannel` msgs is quite bad and it's a bit of a downgrade from the current version of LDK, as the value would then only be available in the `Channel` struct at a later state in the message flow (after the first `ChannelUpdate` msg). 
But the alternative of creating a `Channel::counterparty_forwarding_info` field with dummy values is also bad practice. To avoid assigning dummy values, I guess we could make all fields of `CounterpartyForwardingInfo` optional, that that's not great either. How should this be handled?
1. How should the `htlc_minimum_msat` value passed in the `ChannelUpdate` msg be handled? Should it be ignored like the current version of LDK, or would we like it to update the Channel's value for every `ChannelUpdate` msg sent? (which in practice should be the value sent over and over again)

Reply:
> In the current version of LDK, the Channel::counterparty_htlc_minimum_msat value is a non optional field that is assigned through the OpenChannel msg for inbound channels in the new_from_req function, and through the AcceptChannel msg in the accept_channel function in for outbound channels. The new Channel::counterparty_forwarding_info::outbound_htlc_minimum_msat would be an Optional field that's assigned at a later stage in the message flow.

I'm a bit confused - both should be set on inbound-open/outbound-accept - the reason the new field is optional is the user may have a counterparty object lying around already, but that doesn't impact us. We can set it appropriately in memory in the in-channel object based on its serialization in a different place.

Reply:
Ah! Sorry about the confusion, yea, you're right. GIven there are two *conflicting* fields in the protocol it makes sense, I suppose, to store both.

Reply:
Ah! Sorry for the confusion here, yea. It really sucks the protocol has two somewhat redundant fields for this :(. I think, however, we should set the `outbound_htlc_minimum_msat` field to the maximum of what the peer told us in their init and what's in the channel update.

Reply:
Would the `ChannelUpdate` be more relevant than what was in the `OpenChannel`/`AcceptChannel` message? Just wondering why we would take the max if we later learn a different value.

Reply:
Ok thanks, so would we like to keep the `Channel::counterparty_htlc_minimum_msat` field then as the protocol has two fields for `htlc_minimum_msat` (and maybe rename it to `counterparty_init_htlc_minimum_msat`)?

Or should we just stick to the `counterparty_forwarding_info::outbound_htlc_minimum_msat` field and assign it in when receiving the `OpenChannel`/`AcceptChannel` msgs as well?

Reply:
Okay, I'm an idiot, for the fourth time here.The htlc_min/max included in the channel_update we receive from our peer is re-announcing our limits. The updates can also be things announced publicly and give routing information for how to get from our peer to us, so it has to be the limits we apply on HTLCs. Our current code is correct to ignore them - we set the inbound HTLC limits, not our counterparty.

Reply:
Oh! Ok, then assigning the `counterparty_forwarding_info::outbound_htlc_minimum_msat` based on the `ChannelUpdate` msg like my code currently does is also incorrect. I guess it should be assigned using the same logic as the `get_holder_htlc_maximum_msat` function I've created, but using `counterparty_max_htlc_value_in_flight_msat` instead of `holder_max_htlc_value_in_flight_msat` correct?

Reply:
Hmmm, yea, that's kinda tricky. I definitely think the public API needs to remain the same - `forwarding_info` in the `ChannelDetails` needs to remain `None` until we have the feerate info from our counterparty. How that translates to storage inside the `Channel`, though, I don't have a super strong opinion. Another thing to think about is maybe moving the new fields to `ChannelCounterparty` instead of `CounterpartyForwardingInfo`. This would line up a bit better with the docs on `ChannelCounterparty::forwarding_info` anyway, which says "Information on the fees and requirements that the counterparty requires when forwarding payments to us through this channel" - the new fields are about forwarding outbound from us.

Reply:
Ok thanks! Yes definitely makes sense that we don't want to change the public API. 

Great, removing `counterparty_htlc_minimum_msat` from the `forwarding_info` would help a lot. Would we be ok with keeping the `Channel::counterparty_htlc_minimum_msat` field in the Channel struct then, and use the public functions of the `Channel` to fetch that value when creating the `ChannelCounterparty` in `ChannelManager::list_channels_with_filter`, to assign the new `ChannelCounterparty::outbound_htlc_minimum_msat` field?

Reply:
Yep, sounds good.